staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -488,6 +488,18 @@ func (c completedConfig) New(name string, delegationTarget DelegationTarget) (*G
 			return nil, err
 		}
 	}
+
+	auditHookName := "audit-backend"
+	if c.AuditBackend != nil && !s.isPreShutdownHookRegistered(auditHookName) {
```

Comment:
why this `isPreShutdownHookRegistered` call? @deads2k do you remember why this was added above?

Reply:
`	if c.SharedInformerFactory != nil && !s.isPostStartHookRegistered(genericApiServerHookName) {
`
like above, because this func can be called many times when boot

Reply:
`apiextensions-apiserver` also call this `New`

Reply:
> because this func can be called many times when boot

I don't think we suppport calling it many times.